activemodel/lib/active_model/attribute_methods.rb

```diff
@@ -213,6 +213,8 @@ def eagerly_generate_alias_attribute_methods(new_name, old_name) # :nodoc:
       end
 
       def generate_alias_attribute_methods(code_generator, new_name, old_name)
+        raise ActiveModel::MissingAttributeError, "missing attribute #{old_name} for #{self.class}" unless has_attribute?(old_name)
```

Comment:
What do you think of having a longer error? Something like "#{old_name} is not an Active Record attribute on {self.class}. alias_attribute cannot be used with plain methods or non-databased backed accessors; you likely want to use alias instead."

It gets kind of wordy and I dunno if the advice will be more confusing.

Reply:
👍 Yeah, I think something more descriptive is good. Maybe even `alias_attribute cannot be used with non-attribute method #{old_name}`